app/models/user.rb

```diff
@@ -1,8 +1,17 @@
 class User < ApplicationRecord
-  has_secure_password
+  has_secure_password validations: false, if: "uid"
+  validates :email, confirmation: true, if: "uid.nil?", on: :create
```

Comment:
let's do a uniqueness validation on emails.  I don't think we'll need any validations on the rest of them (uid provided by github and oauth_token provided by oauth).  The other things are out of the users control.